apps/fz_http/priv/repo/migrations/20221201216913_create_gateway_firewall_policies.exs

```diff
@@ -0,0 +1,68 @@
+defmodule FzHttp.Repo.Migrations.CreateGatewayFirewallPolicies do
+  use Ecto.Migration
+
+  @create_query "CREATE TYPE protocol_enum AS ENUM ('tcp', 'udp')"
+  @drop_query "DROP TYPE protocol_enum"
+
+  def change do
+    execute(@create_query, @drop_query)
+
+    create table(:gateway_firewall_policies, primary_key: false) do
+      add(:id, :uuid, primary_key: true, default: fragment("gen_random_uuid()"))
+      add(:gateway_id, references(:gateways, on_delete: :delete_all, type: :uuid), null: false)
+      add(:user_id, references(:users, column: :uuid, type: :uuid))
+      add(:destination, :inet, null: false)
+      add(:port_range, :int4range)
+      add(:protocol, :protocol_enum)
+
+      timestamps(type: :utc_datetime_usec)
+    end
+
+    create(
+      constraint("gateway_firewall_policies", :port_range_is_within_valid_values,
+        check: "port_range <@ int4range(1, 65535)"
+      )
+    )
+
+    create(
+      constraint("gateway_firewall_policies", :port_range_has_optional_protocol,
+        check: "port_range IS NOT NULL AND protocol IS NOT NULL OR protocol IS NULL"
+      )
+    )
+
+    execute("""
+    CREATE OR REPLACE FUNCTION notify_gateway_firewall_policy_changes()
+    RETURNS trigger AS $$
+    DECLARE
+      row record;
+    BEGIN
+      row := NEW;
+
+      IF (TG_OP = 'DELETE') THEN
+        row := OLD;
+      END IF;
+
+      PERFORM pg_notify(
+        'gateway_firewall_policies_changed',
+        json_build_object(
+          'op', TG_OP,
+          'row', row_to_json(row)
+        )::text
+      );
+
+      RETURN row;
+    END;
+    $$ LANGUAGE plpgsql;
+    """)
+
+    execute("""
+    CREATE CONSTRAINT TRIGGER gateway_firewall_policies_changed
+    AFTER INSERT OR DELETE ON gateway_firewall_policies
```

Comment:
Do we need `UPDATE` here as well?

Reply:
Yes, it was left out because the rules are either added/deleted but never updated. This'll probably change and better to address it now by adding an `UPDATE` notification here as well.

Reply:
However, hmm, if we wanted to audit a history of rules would we ever update the record and instead just insert a new one and invalidate the old row somehow?

Reply:
Or rolling back rules would be a better example of keeping old rules around.

Reply:
This is better explored in future features. Resolving as I've added an update trigger.